File: Upcoder/Lập trình cơ bản/5.Mảng/Mang ZicZac cơ bản/ZicZacCBan.py
# Đọc n và m từ đầu vào
n, m = map(int, input().split())

# Tạo mảng 2 chiều và điền các phần tử
a = []
temp = 1
for i in range(n):
    row = []
    for j in range(m):
        row.append(temp)
        temp += 1
    a.append(row)

# In mảng theo thứ tự yêu cầu
c = 1
for i in range(n):
    if c % 2 != 0:  # Hàng lẻ: in từ trái sang phải
        for j in range(m):
            print(a[i][j], end=' ')
    else:  # Hàng chẵn: in từ phải sang trái
        for j in range(m - 1, -1, -1):
            print(a[i][j], end=' ')
    c += 1
    print()

File: Upcoder/Lập trình cơ bản/5.Mảng/Mang ZicZac cơ bản/ZicZacCBan.cs
using System;

namespace Mang2c
{
    public class Program
    {
        static void Main(string[] args)
        {
            // Đọc n và m từ đầu vào
            string[] input = Console.ReadLine().Split(' ');
            int n = int.Parse(input[0]);
            int m = int.Parse(input[1]);
            
            int[,] a = new int[n, m];
            int temp = 1;
            // Nhập các phần tử của mảng
            for (int i = 0; i < n; i++)
            {
                for (int j = 0; j < m; j++)
                {
                    a[i, j] = temp;
                    temp ++;
                }
            }

            int c = 1;
            for (int i = 0; i < n; i++)
            {
                if (c % 2 != 0) 
                {
                    for (int j = 0; j < m; j++)
                        Console.Write("{0} ", a[i, j]);
                }
                else 
                {
                    for (int j = m - 1; j >= 0; j--)
                        Console.Write("{0} ", a[i, j]);
                }
                c++;
                Console.WriteLine();
            }
        }
    }
}
